src/vampyr/trees/trees.h

```diff
@@ -124,140 +177,47 @@ template <int D> void trees(pybind11::module &m) {
-                out->multiply(1.0, *inp);
-                return out;
-            },
-            py::is_operator())
-        .def(
-            "__truediv__",
-            [](FunctionTree<D> *inp, double c) {
-                auto out = std::make_unique<FunctionTree<D>>(inp->getMRA());
-                FunctionTreeVector<D> vec;
-                vec.push_back({1.0 / c, inp});
-                build_grid(*out, vec);
-                add(-1.0, *out, vec);
-                return out;
-            },
-            py::is_operator())
-        .def(
-            "__itruediv__",
-            [](FunctionTree<D> *out, double c) {
-                out->rescale(1.0 / c);
-                return out;
-            },
-            py::is_operator())
-        .def(
-            "__pow__",
-            [](FunctionTree<D> *inp, double c) {
-                auto out = std::make_unique<FunctionTree<D>>(inp->getMRA());
-                copy_grid(*out, *inp);
-                copy_func(*out, *inp);
-                refine_grid(*out, 1);
-                out->power(c);
-                return out;
-            },
-            py::is_operator())
-        .def(
-            "__ipow__",
-            [](FunctionTree<D> *out, double c) {
-                refine_grid(*out, 1);
-                out->power(c);
-                return out;
-            },
-            py::is_operator());
+        .def("__pos__", &impl__pos__<D>, py::is_operator())
+        .def("__neg__", &impl__neg__<D>, py::is_operator())
+        .def("__add__", py::overload_cast<FunctionTree<D> *, FunctionTree<D> *>(&impl__add__<D>), py::is_operator())
+        .def("__iadd__", py::overload_cast<FunctionTree<D> *, FunctionTree<D> *>(&impl__add__<D>), py::is_operator())
+        .def("__sub__", py::overload_cast<FunctionTree<D> *, FunctionTree<D> *>(&impl__sub__<D>), py::is_operator())
+        .def("__isub__", py::overload_cast<FunctionTree<D> *, FunctionTree<D> *>(&impl__sub__<D>), py::is_operator())
+        .def("__mul__", py::overload_cast<FunctionTree<D> *, double>(&impl__mul__<D>), py::is_operator())
+        .def("__rmul__", py::overload_cast<FunctionTree<D> *, double>(&impl__mul__<D>), py::is_operator())
+        .def("__imul__", py::overload_cast<FunctionTree<D> *, double>(&impl__mul__<D>), py::is_operator())
+        .def("__mul__", py::overload_cast<FunctionTree<D> *, FunctionTree<D> *>(&impl__mul__<D>), py::is_operator())
+        .def("__imul__", py::overload_cast<FunctionTree<D> *, FunctionTree<D> *>(&impl__mul__<D>), py::is_operator())
+        .def("__truediv__", py::overload_cast<FunctionTree<D> *, double>(&impl__truediv__<D>), py::is_operator())
+        .def("__itruediv__", py::overload_cast<FunctionTree<D> *, double>(&impl__truediv__<D>), py::is_operator())
+        .def("__pow__", py::overload_cast<FunctionTree<D> *, double>(&impl__pow__<D>), py::is_operator())
+        .def("__ipow__", py::overload_cast<FunctionTree<D> *, double>(&impl__pow__<D>), py::is_operator());
```

Comment:
`__truediv__`, `__itruediv__`, `__pow__` and `__ipow__` do not need `overload_cast`.